cmd/erasure-sets.go

```diff
@@ -508,9 +508,18 @@ func (s *erasureSets) cleanupDeletedObjects(ctx context.Context) {
 			// Reset for the next interval
 			timer.Reset(globalAPIConfig.getDeleteCleanupInterval())
 
+			var wg sync.WaitGroup
 			for _, set := range s.sets {
-				set.cleanupDeletedObjects(ctx)
+				wg.Add(1)
```

Comment:
We can move the check for `set == nil` to before calling the goroutine.

Reply:
Yeah well it never really happens so its just defensive and its a no-op anyways for waitgroup point of view. So just leaving it as is.